src/scripts/quests/questTypes/DefeatDungeonBossQuest.ts

```diff
@@ -0,0 +1,59 @@
+class DefeatDungeonBossQuest extends Quest implements QuestInterface {
+    initialValue?: number;
+    customReward?: () => void;
+    target: PokemonNameType;
+    targetEncountered: boolean;
+
+    constructor(public dungeon: string, private tmpTarget : string, reward: (() => void) | number = 0, description?: string, onLoad?: (() => void)) {
+        const qpReward = typeof reward == 'number' ? reward : 0;
+        super(1, qpReward);
+        this.customDescription = description ?? `Defeat ${this.tmpTarget} in ${this.dungeon}.`;
+        this.customReward = typeof reward == 'function' ? reward : undefined;
+        this._onLoad = typeof onLoad == 'function' ? onLoad : undefined;
+        this.targetEncountered = false;
+
+        // Target is undefined until onLoad is called. TmpTarget keeps track of raw data.
+        this.focus = ko.pureComputed(() => {
+            const enemyPokemon = Battle.enemyPokemon();
+            const defeatedBoss = DungeonRunner.defeatedBoss();
+            if (enemyPokemon) {
+                if (enemyPokemon.name === this.target
+                    && DungeonRunner.dungeon.name === this.dungeon
+                    && App.game.gameState === GameConstants.GameState.dungeon) {
+                    this.targetEncountered = true;
```

Comment:
I don't like pure computed setting other values. Those should be pure computed too